ledger/complete/ledger.go

```diff
@@ -434,14 +434,15 @@ func (l *Ledger) ExportCheckpointAt(
 	l.logger.Info().Msgf("finished running pre-checkpoint reporters")
 
 	l.logger.Info().Msg("creating a checkpoint for the new trie")
-	writer, err := realWAL.CreateCheckpointWriterForFile(outputDir, outputFile, &l.logger)
-	if err != nil {
-		return ledger.State(hash.DummyHash), fmt.Errorf("failed to create a checkpoint writer: %w", err)
-	}
 
 	l.logger.Info().Msg("storing the checkpoint to the file")
 
-	err = realWAL.StoreCheckpoint(writer, newTrie)
+	err = os.MkdirAll(outputDir, os.ModePerm)
+	if err != nil {
+		return ledger.State(hash.DummyHash), fmt.Errorf("could not create output dir %v: %w", outputDir, err)
```

Comment:
A general nit: you are using `%v` a lot which may prevent static analyzer from detecting a class of bugs when a programmer accidentally passes a different variable than they actually wanted to log.